pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -182,14 +182,11 @@ def save_checkpoint(self, trainer, pl_module):
         This method runs on all ranks, it is the responsibility of `self.save_function`
         to handle correct behaviour in distributed training, i.e., saving only on rank 0.
         """
-        epoch = trainer.current_epoch
-
         if (
             self.save_top_k == 0  # no models are saved
             or self.period < 1  # no models are saved
-            or (epoch + 1) % self.period  # skip epoch
+            or (trainer.current_epoch + 1) % self.period  # skip epoch
             or trainer.running_sanity_check  # don't save anything during sanity check
-            or self.epoch_last_check == epoch  # already saved
```

Comment:
@Borda when removing this line, we now need to take care of the manual calling in trainer/training_loop.py Line 200. 
Otherwise, on_validation_end gets called twice (once manually and once through callback system) and we save the same checkpoint twice, resulting in a v1 in the name.

Reply:
The best is to remove this manual invokation (there is a TODO anyway).

Reply:
This is the line
https://github.com/PyTorchLightning/pytorch-lightning/blob/0838c6bfce0b76cd4c660807333814cafab8cd14/pytorch_lightning/trainer/training_loop.py#L200

Reply:
well we can rather replace by calling just once per step, right?

Reply:
I see, so basically, we want to have a checkpointing method wich force to save the last state...
just thinking that as a user I would expect to have always saved last just in case the training accidentally crash...

Reply:
true, though I think we need to address this in a separate PR.
let's keep it like you have right now. I think it's good :)